### PR TITLE
Implement Effective HP and refactor defence calculations to be more accurate

### PR DESCRIPTION
**Pull request by Regisle**
This PR aims to rework a large section of calcs defense, to improve the accuracy of the maximum hit, and dot/hit "EHP" values as well as allow for more indepth enemy damage calculations

before:
image
after: (note that config values are not the same as the 1 "enemyDamage" input has been split into each of its specific elements)
image

done:
    split enemy damage input
    split guard and aegis values
    fixed dot total pool and dot ehp
    moved and cleaned up some code
   modify enemy input damage based on modifiers to enemy damage (eg enfeeble curse)
   add presets for enemy damage based on boss config
   flat damage taken
   add enemy hitrate and player survival time
   ward
todo:
   frostshield
   breakdowns
   formatting
   code cleanup
Extensions:
   sorting
   maximum hit taken rework

### CODE REVIEW

**Comment by Wires77** [comment]
Where does this base value come from?  I have 10k set in the configuration:
image

Having zeroes in each place for the elements may be confusing, I'd copy them across or leave them completely empty.  Also put "Mind Over Matter %" to make that clear:
image

File: src/Modules/ModParser.lua [hunk 1]
```diff
@@ -1540,6 +1545,9 @@ local specialModList = {
 		mod("EnemyModifier", "LIST", { mod = mod("PhysicalDamageTaken", "INC", num) }),
 		mod("EnemyModifier", "LIST", { mod = mod("FireDamageTaken", "INC", num) }),
 	} end,
+	["enemies near your totems deal (%d+)%% less damage"] = function(num) return {
```
Comment (Wires77) [inline src/Modules/ModParser.lua:1548]: This line can probably go in the `preFlagList` as "enemies near your totems deal".

**Review by Wires77 — COMMENTED** [review]
Still going through the code changes, but the UI looks pretty good.  This dropdown and tooltip is meaningless, however:
image

File: src/Modules/ModParser.lua (hunk 1 above)
Comment (Regisle) [reply src/Modules/ModParser.lua]: moved

**Review by Wires77 — APPROVED** [review]

**Pull request merged**